tests/app/core/models/session-model.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { assertSessionDataIntegrity } from '../../../../src/app/core/models/session.model';
+import { of } from 'rxjs';
+
+const validSessionData = {
+  openPhases: [
+    'phaseBugReporting',
+    'phaseTeamResponse',
+    'phaseTesterResponse',
+    'phaseModeration',
+  ],
+  phaseBugReporting: 'bugreporting',
+  phaseTeamResponse: 'pe-results',
+  phaseTesterResponse: 'testerresponse',
+  phaseModeration: 'pe-evaluation',
+};
+
+describe('Session Model', () => {
+  describe('Assert Session Data Integrity', () => {
+    it('should throw error on unavailable session', () => {
+      of(undefined)
+        .pipe(assertSessionDataIntegrity())
+        .subscribe({
+          error: (err) =>
+            expect(err).toEqual(new Error('Session Data Unavailable')),
+        });
+    });
+    it('should throw error on incorrect session data', () => {
+      of({ key: '' })
+        .pipe(assertSessionDataIntegrity())
+        .subscribe({
+          error: (err) =>
+            expect(err).toEqual(
+              new Error('Session Data is Incorrectly Defined')
+            ),
+        });
+      of({ key: undefined })
+        .pipe(assertSessionDataIntegrity())
+        .subscribe({
+          error: (err) =>
+            expect(err).toEqual(
+              new Error('Session Data is Incorrectly Defined')
+            ),
+        });
+    });
+    it('should throw error on session with no open phases', () => {
+      of({ openPhases: [] })
+        .pipe(assertSessionDataIntegrity())
+        .subscribe({
+          error: (err) =>
+            expect(err).toEqual(new Error('There are no accessible phases.')),
+        });
+    });
+    it('should pass valid session data', () => {
+      of(validSessionData)
+        .pipe(assertSessionDataIntegrity())
+        .subscribe((el) => expect(el).toEqual(validSessionData));
+    });
```

Comment:
So I think this is okay but technically in the actual use-case, students should only receive an `openPhases` array with a single element inside. Should the `validSessionData` also contain a single open phase? So that this sort of simulates that particular use-case as opposed to the "dev's" use case.

What do you think? I mean it works either way so its no issue.

Reply:
I think this is fine - no need to specifically test the case where there is one open phase.

(Anyway, the app itself should still work if multiple phases are open. It's up to the user to restrict the number of simultaneously open phases)